src/diffusers/models/unet_2d_condition.py

```diff
@@ -651,11 +656,20 @@ def forward(
             logger.info("Forward upsample size to force interpolation output size.")
             forward_upsample_size = True
 
-        # prepare attention_mask
+        # ensure attention_mask is a bias, and make it broadcastable over multi-head-attention channels
         if attention_mask is not None:
+            # assume that mask is expressed as:
```

Comment:
That's a very nice addition to the comment :heart:

Reply:
I've also just now revised the explanation above about the purpose of the singleton dimension. looks like it wasn't for broadcasting over channels, but rather for broadcasting over query_tokens.